packages/react/src/components/checkbox/Checkbox.tsx

```diff
@@ -81,7 +81,7 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
           aria-describedby={errorText ? `${id}-error` : null}
           {...rest}
         />
-        <label htmlFor={id} className={classNames(styles.label)}>
+        <label aria-hidden={!label} htmlFor={id} className={classNames(styles.label)}>
```

Comment:
This new property in Checkbox needs to be described in checkbox documentation, when it should be used etc.

Reply:
This does not change the direct react component api, so not sure if we need to add it to documentation. 🤔 

It does change the behaviour of the component little though; in situations where the user of the checkbox has not provided visible label, it will hide the (empty) label tag from screenreaders. This change basically makes it possible to not provide label for the checkbox, as earlier a11y tools would complain that empty labels are violating a11y. Why not just remove the whole label element then (when user does not give label)? Problem was that our Checkbox component was programmed in a way that it requires the label tag. If you remove the whole label tag, the checkbox disappears 😅. If we wish to support a checkbox without a label, we might have to make bigger changes to the existing Checkbox implementation.

Reply:
- [ ] Remove unnecessary changes to Checkbox component

Reply:
I think the checkbox is defined to have a mandatory label and tables are the first exception to that. It was the biggest reason we had a separate ticket for making one without label to be used inside other elements.